ktor-server/ktor-server-cio/common/src/io/ktor/server/cio/UnixSocketConnectorConfig.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2014-2025 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.cio
+
+import io.ktor.server.engine.ApplicationEngine
+import io.ktor.server.engine.ConnectorType
+import io.ktor.server.engine.EngineConnectorBuilder
+import io.ktor.server.engine.EngineConnectorConfig
+
+/**
+ * Represents a Unix domain socket connector configuration.
+ *
+ * [Report a problem](https://ktor.io/feedback/?fqname=io.ktor.server.engine.UnixSocketConnectorConfig)
+ */
+public interface UnixSocketConnectorConfig : EngineConnectorConfig {
+    /**
+     * Path to the Unix domain socket file.
+     *
+     * [Report a problem](https://ktor.io/feedback/?fqname=io.ktor.server.engine.UnixSocketConnectorConfig.socketPath)
+     */
+    public val socketPath: String
+}
+
+/**
+ * Mutable implementation of UnixSocketConnectorConfig for building connectors programmatically
+ *
+ * [Report a problem](https://ktor.io/feedback/?fqname=io.ktor.server.engine.UnixSocketConnectorBuilder)
+ */
+public class UnixSocketConnectorBuilder : EngineConnectorBuilder(ConnectorType.UNIX), UnixSocketConnectorConfig {
```

Comment:
It seems that we can remove inheritance from `EngineConnectorBuilder` here and keep only `socketPath` field in the builder.

Reply:
let me check

Reply:
I've just realised that it makes sense only if we have a separate interface for `UnixSocketConnectorConfig` which doesn't contain extra fields. So it is connected to this comment - https://github.com/ktorio/ktor/pull/4844#discussion_r2081293565

Reply:
As a possible solution:
```kotlin
// Doesn't implement `EngineConnectorConfig`
public interface UnixSocketConnectorConfig {
    public var socketPath: String
}

// Public-facing builder interface
public interface UnixSocketConnectorBuilder : UnixSocketConnectorConfig {
    public var socketPath: String
}

// Internal implementation of `EngineConnectorBuilder` and `UnixSocketConnectorBuilder`
// Also implements `EngineConnectorConfig` so can be added to connectors list
internal class UnixSocketConnectorBuilderImpl : EngineConnectorBuilder(ConnectorType.UNIX), UnixSocketConnectorBuilder {
    override var socketPath: String = "/tmp/ktor.sock"
}
```